api/handlers.go

```diff
@@ -31,3 +32,124 @@ func (api *API) userHandlerGET(w http.ResponseWriter, req *http.Request, _ httpr
 	}
 	WriteJSON(w, u)
 }
+
+// userUploadsHandler returns all uploads made by the current user.
+func (api *API) userUploadsHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	sub, _, _, err := tokenFromContext(req)
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	u, err := api.staticDB.UserBySub(req.Context(), sub)
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	ups, err := api.staticDB.UploadsByUser(req.Context(), *u)
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
+	}
+	WriteJSON(w, ups)
+}
+
+// userDownloadsHandler returns all downloads made by the current user.
+func (api *API) userDownloadsHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	sub, _, _, err := tokenFromContext(req)
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	u, err := api.staticDB.UserBySub(req.Context(), sub)
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	ups, err := api.staticDB.DownloadsByUser(req.Context(), *u)
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
+	}
+	WriteJSON(w, ups)
+}
+
+// trackUploadHandler registers a new upload in the system.
+func (api *API) trackUploadHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	sl := req.PostFormValue("skylink")
+	if sl == "" {
+		WriteError(w, errors.New("missing parameter 'skylink'"), http.StatusBadRequest)
+		return
+	}
+	sub, _, _, err := tokenFromContext(req)
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	skylink, err := api.staticDB.Skylink(req.Context(), sl)
+	if err == database.ErrInvalidSkylink {
+		WriteError(w, err, http.StatusBadRequest)
+		return
+	}
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	u, err := api.staticDB.UserBySub(req.Context(), sub)
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	_, err = api.staticDB.UploadCreate(req.Context(), *u, *skylink)
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	if skylink.Size == 0 {
```

Comment:
Should this be

```suggestion
	if skylink.Size != 0 {
```

?

Reply:
Nevermind! The name `Size` confused me.

Reply:
I added a little more info in the comment. Hopefully that would make it a little less confusing.